sdk/core/System.ClientModel/src/ModelReaderWriter/ModelJsonConverter.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// A generic converter which allows <see cref="JsonSerializer"/> to be able to write and read any models that implement <see cref="IJsonModel{T}"/>.
+/// </summary>
+#if !NET5_0 // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
+[RequiresUnreferencedCode("The constructors of the type being deserialized are dynamically accessed and may be trimmed.")]
+#endif
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+public class ModelJsonConverter : JsonConverter<IJsonModel<object>>
+#pragma warning restore AZC0014 // Avoid using banned types in public API
+{
+    /// <summary>
+    /// Gets the <see cref="ModelReaderWriterOptions"/> used to read and write models.
+    /// </summary>
+    public ModelReaderWriterOptions Options { get; }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="ModelJsonConverter"/> with a default options of <see cref="ModelReaderWriterOptions.Json"/>.
+    /// </summary>
+    public ModelJsonConverter()
+        : this(ModelReaderWriterOptions.Json) { }
+
+    /// <summary>
+    /// Initializes a new instance of <see cref="ModelJsonConverter"/>.
+    /// </summary>
+    /// <param name="options">The <see cref="ModelReaderWriterOptions"/> to use.</param>
+    public ModelJsonConverter(ModelReaderWriterOptions options)
+    {
+        Options = options;
+    }
+
+    /// <inheritdoc/>
+    public override bool CanConvert(Type typeToConvert)
+    {
+        return !Attribute.IsDefined(typeToConvert, typeof(JsonConverterAttribute));
+    }
+
+    /// <inheritdoc/>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+    public override IJsonModel<object> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
It might be good to validate the interplay of STJ options and MRW options, e.g. in test coverage. 

One question I have around how options work together is whether it's better to fail silently, e.g. if we can't apply STJ options because it would break MRW contracts, etc, or throw in those cases to notify the end user.  If we later wanted to apply an STJ option, would it be a breaking change to change what MRW does?

Reply:
Added a case to validate pretty print options are honored and structural / naming options are not since mrw owns the shape on the wire if you are deferring to mrw.

Reply:
Are there other things we might care about on JsonSerializerOptions?  I am peeking at it now and there is a lot there, like different ways to handle numbers, comments, commas, type resolution, etc.  I don't have a good sense of scenarios surrounding use for those, but if someone expected to be able to control type serialization, e.g., and it didn't happen because MRW took over, would that be surprising or result in bad outcomes for end-users?  Would it be worth documenting that JsonSerializationOptions on input to Read or Write is ignored by the converter?

Reply:
added comment in the type remarks to indicate that only indentation settings are honored.